### ISSUE TITLE
ERROR: ProcessNewBlock : AcceptBlock FAILEDERROR: CheckProofOfStake() : INFO: read txPrev failed for input 1WARNING: ProcessBlock(): check proof-of-stake failed for block

### ISSUE THREAD

**Issue by benzmuircroft** [DiviProject/Divi#226]
Ubuntu 20
Divi Core Daemon version v2.5.1

I threw away in the trash all the files in datadir except for wallet.dat and divi.conf then restarted and came back a few days later to see it is stuck again.

```
rpcuser=redacted
rpcpassword=redacted

prune=2048
txindex=0
blocksonly=1
dbcache=500
maxsigcachesize=500
maxconnections=13
rpcthreads=5
maxorphantx=10

port=51472
rpcport=51473
listen=1

datadir=/home/DIVI/data

walletnotify=[ -e "/home/DIVI/tx.js" ] && node /home/DIVI/tx.js %s
blocknotify=[ -e "/home/DIVI/block.js" ] && node /home/DIVI/block.js %s

#https://chainz.cryptoid.info/divi/#!network
addnode=192.241.192.197
addnode=137.184.70.30
addnode=159.223.164.75
addnode=159.89.165.165
addnode=128.199.106.227
addnode=134.209.208.96
addnode=138.197.66.153
addnode=139.59.70.25
addnode=143.110.251.227
addnode=157.230.88.148
addnode=159.89.122.69
addnode=167.71.237.62
addnode=46.101.98.223
addnode=136.55.113.42
addnode=159.223.24.39
addnode=165.227.215.202
addnode=174.193.202.242
addnode=178.62.198.111
addnode=188.166.95.203
addnode=207.154.233.230
addnode=217.87.113.161
addnode=46.101.87.198
addnode=64.225.61.24
addnode=66.74.205.142
addnode=77.13.74.238
addnode=84.115.216.42
addnode=97.93.124.235
```

```
2021-12-12 17:31:26 ERROR: ProcessNewBlock : AcceptBlock FAILEDERROR: CheckProofOfStake() : INFO: read txPrev failed for input 1WARNING: ProcessBlock(): check proof-of-stake failed for block eecbfa137f2192888fb4abf7181d065273989d0f1d106e28d42cca79626b37d6
2021-12-12 17:31:26 ERROR: ProcessNewBlock : AcceptBlock FAILEDERROR: CheckProofOfStake() : INFO: read txPrev failed for input 1WARNING: ProcessBlock(): check proof-of-stake failed for block eecbfa137f2192888fb4abf7181d065273989d0f1d106e28d42cca79626b37d6
2021-12-12 17:31:26 ERROR: ProcessNewBlock : AcceptBlock FAILEDsocket send error Connection reset by peer (104)
2021-12-12 17:35:58 connect() to 37.235.55.45:51472 failed after select(): Connection refused (111)
2021-12-12 17:35:58 receive version message: DIVI Core: 2.0.1.0: version 70915, blocks=1679594, us=15.235.26.150:50826, peer=1862
2021-12-12 17:36:03 socket sending timeout: 1201s
2021-12-12 17:36:03 receive version message: DIVI Core: 2.0.1.0: version 70915, blocks=1679594, us=15.235.26.150:39488, peer=1863
2021-12-12 17:36:25 ERROR: CheckProofOfStake() : INFO: read txPrev failed for input 1WARNING: ProcessBlock(): check proof-of-stake failed for block eecbfa137f2192888fb4abf7181d065273989d0f1d106e28d42cca79626b37d6
2021-12-12 17:36:25 ERROR: ProcessNewBlock : AcceptBlock FAILEDERROR: CheckProofOfStake() : INFO: read txPrev failed for input 1WARNING: ProcessBlock(): check proof-of-stake failed for block eecbfa137f2192888fb4abf7181d065273989d0f1d106e28d42cca79626b37d6
2021-12-12 17:36:53 ERROR: ProcessNewBlock : AcceptBlock FAILED
```

**Comment by galpHub** [comment]
TLDR: set txindex=1 in the config to fix.

Hey there,
I was able to reproduce your experience and the issue in question boils down to a minor bug that is triggered when txindex is disabled. That will be resolved in an upcoming release. Sorry for the inconvenience!

Best regards and happy staking!

**Comment by benzmuircroft** [comment]
ah. well I will be unable to integrate your coin in my application until this fix is applied. As I rely on having the txindex off in my code. 

Good luck, hurry and do quick!
